pkgs/os-specific/linux/apparmor/default.nix

```diff
@@ -249,8 +274,32 @@ let
     meta = apparmor-meta "kernel patches";
   };
 
+  # Generate generic AppArmor rules in a file,
+  # from the closure of given rootPaths.
+  # To be included in an AppArmor profile like so:
+  # include "$(apparmorRulesFromClosure {} [pkgs.hello]}"
+  apparmorRulesFromClosure =
+    { # The store path of the derivation is given in $path
+      additionalRules ? []
+      # TODO: factorize here some other common paths
+      # that may emerge from use cases.
+    , baseRules ? [
+        "r $path"
+        "r $path/etc/**"
+        "r $path/share/**"
+        # Note that not all libraries are prefixed with "lib",
+        # eg. glibc-2.30/lib/ld-2.30.so
+        "mr $path/lib/**.so*"
+        # eg. glibc-2.30/lib/gconv/gconv-modules
+        "r $path/lib/**"
+      ]
+    }: rootPaths: runCommand "apparmor-closure-rules" {} ''
```

Comment:
I think these should be named by argument, for clarity in the store

Reply:
Do you mean adding something in "apparmor-closure-rules"? what would be that something?

Reply:
Yes, I mean pass in a designation as an argument to apparmorRulesFromClosure, so that there is some context rather than hundreds of paths designated "apparmor-closure-rules"

Reply:
I've added an optional `name` attribute.